hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -175,7 +190,12 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           EnvironmentContext envContext = new EnvironmentContext(
               ImmutableMap.of(StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE)
           );
-          client.alter_table(database, tableName, tbl, envContext);
+
+          try {
+            alterTableMethod.invoke(client, database, tableName, tbl, envContext);
+          } catch (Exception e) {
```

Comment:
Try to avoid catching `Exception` so that this doesn't swallow unexpected errors.